### ISSUE TITLE
Revert the change to the miss chance formula

### ISSUE THREAD

**Issue by ohganot** [LightsHope/server#844]
**Reasoning**

- We have a source from TBC with a clear a well-tested formula, which doesn't seem to conflict with any known vanilla mechanic
http://web.archive.org/web/20090715100718/http://elitistjerks.com/f31/t11885-rogue_warrior_weapon_skill_adjustment_discussion/p20/#post469576
- We have a screenshot from Vanilla with a rogue missing with 8% hit chance http://web.archive.org/web/20081111013822/http://sp00n.pytalhost.com/misc/backstabmiss.jpg
- The current formula results in 5.6% miss chance vs a lvl 63 mob when you have 315 weapon skill. This is clearly not accurate, since base miss chance should be 5%
- The reasoning for the previous change was to reduce the huge bump from 304 to 305 weapon skill, but there's no reason to do that, since the biggest benefit of +weapon skill is the increased damage on glancing blows, not the reduced miss chance.

**Comment by fgsfdss** [comment]
#755 for reference.

**Comment by henryboston** [comment]
` float missChance = 5.60f; // The base chance to miss is 5.60%`

The current base miss chance is set to 5.6%. It should be 5% as noted from several old Elitist Jerk theory crafting threads.

**Comment by thepparker** [comment]
The current formula is not quite correct, however neither was the previous formula. I am open to further modifications should we identify one that correctly yields 5% with equal weapon skill, and 8.6% vs level 63 targets with no weapon skill, which seems to be mostly true to Blizzard's implementation.

**Comment by fgsfdss** [comment]
If the only proper research available is the EJ thread and that is what is being used for reference, how did we end up with 8.6% instead of 9?
How credible are the sources stating the former compared to the EJ thread, even despite that being from TBC?
Lastly, is that evidence more valid than that for the base miss chance being 5%, and which is more important?

**Comment by ohganot** [comment]
@thepparker Why do you think that exactly 8.6% was mostly true? Got any sources I haven't managed to find?

Also, in that case, we could just combine the TBC and Light's Hope formula:
https://gyazo.com/6b4fd2b80086747b0632519e747a590f

So something like
```
skillDiff = (BossDefenseSkill - PlayerWeaponSkill)
if (skillDiff > 10)
        misschance = 6.6f + (skillDiff+10) * 0.4f; 
else
        misschance = 5f + skillDiff*0.1f;

// for lvl 60 vs 60
misschance = 5f + skillDiff*0.04f;
```

**Comment by henryboston** [comment]
If we can all agree on the two following truths:
- Base miss chance is 5% versus 60 mob
- Miss chance > 8% versus 63 mob

The formula should be:

```
skillDiff = (BossDefenseSkill - PlayerWeaponSkill)
if (skillDiff > 10)
        misschance = 7.0f + (skillDiff-10) * 0.4f; 
else
        misschance = 5.0f + skillDiff*0.1f;
```

This results in the following data versus level 63 (315 defense skill):
| Skill| Miss % |
| --- | --- |
| 300 | 9% |
| 300 | 9% |

**Comment by ohganot** [comment]
Yes, that is the TBC formula, which I also prefer.

**Comment by thepparker** [comment]
- We know that 5% is the minimum.
- We know that an actual hit chance of 9% (on the player) is the cap.
- We know that 8% is not the cap, since there is footage of players missing with 8% hit chance.
- We are fairly confident that Hunters share hit cap with melee (and that is how it is implemented currently).
- There are many references to hunters requiring 8.6% hit chance to be capped for Tranq shot rotations (and some mentions of requiring 9% in BC, which brings us back to the 9% formula).
- Googling 8.6% hit cap in Vanilla will yield a few posts, though they may just be misinformation built on from previously bad knowledge.
- The only major difference is that the gain from 304->305 is 1% hit rather than 1.4%, so we do still see the much exclaimed 'bump' though to a slightly lesser extent. A question to ask regarding this is: was this actually tested and calculated to be 1.4%, or were the EJ researchers basing it on their formulae?

**Comment by Cephel** [comment]
Since the team seems to pick this up again. I will post an alternative formula that I developed a while ago that hits all the breakpoints, but does not need to have this "bump" that has no proof for it (except from TBC):

| Skill | Miss Chance |
| ---- | ------------- |
| 300 | 8.60% |
| 301 | 8.20% |
| 302 | 7.80% |
| 303 | 7.40% |
| 304 | 7.00% |
| 305 | 6.60% |
| 306 | 6.44% |
| 307 | 6.28% |
| 308 | 6.12% |
| 309 | 5.96% |
| 310 | 5.80% |
| 311 | 5.64% |
| 312 | 5.48% |
| 313 | 5.32% |
| 314 | 5.16% |
| 315 | 5.00% |

This solves the problem of the 5% missrate at 60, without completely nonsensically ruining the formula at 305. TBC research is almost entirely worthless, because they changed the formula 3 or more times in the run-up to removing it entirely, which results in spectacular failposts such as this one:
![Image](https://i.imgur.com/zfoaEc7.png)

The changes visualized would look like this:
![image](https://user-images.githubusercontent.com/6716857/35193765-e3a416a4-fea7-11e7-9085-598e52889469.png)

The needed formula for this are:
```(5 - (diff * 0.16)) / 100``` for 305 and up
and ```(5.6 - ((diff+10) * 0.4 - 1)) / 100``` 304 and lower values
respective against a 315 target.

This results in the following hitcaps for various weapon and/or racial combinations:
| Skill | Combination | Miss Chance |
| 304 | Maladath | 7.00% |
| 305 | Orc / Human | 6.60% |
| 307 | Edgemaster's | 6.28% |
| 307 | Maladath + Human | 5.96% |
| 307 | Orc / Human + Edgemaster's | 5.48% |
| 315 | Same level | 5.00% |

The resulting formula has the advantage of not relying on buggy TBC research which results in a 1.40% gap in the swap of the weapon formula, still hitting 5.00% as its base value, which was often requested and properly scales weapon skill values.

**Comment by Rapter29** [comment]
I would rather have a documented TBC based formula, than a speculative formula that somebody created. Just because it is within the bounds of certain evidence, doesn't mean that it at all resembles the vanilla formula. The point is to replicate the vanilla experience as closely as possible, based on evidence. Not create a new experience based on a lack of evidence.

**Comment by Cephel** [comment]
There is no legitimate proof for the weapon skill bump and my formula has significantly less impact on the vanilla playing experience than the TBC formula. Both formulas are speculative.

**Comment by VeryRare** [comment]
@thepparker 

> was this actually tested and calculated to be 1.4%, or were the EJ researchers basing it on their formulae?

From the EJ posts I read it appeared that the 'gap' is based off their formula.

I don't see why the 'gap' is such a problem. From a game design standpoint it makes sense to me, as soon as a mob is level 61 it's considered to be 'raid' level. Level 61 and 305 weapon skill are both big bumps in terms of mob difficulty. 

I don't know if I've ever seen hard proof of a 9% miss chance vs a 8.6% chance. To mean, that is the only question that is left to be unanswered in this discussion. I'm honestly fine with either since effectively they are the same in terms of how much hit gear you need.

**Comment by Cephel** [comment]
>I don't see why the 'gap' is such a problem. From a game design standpoint it makes sense to me, as soon as a mob is level 61 it's considered to be 'raid' level. Level 61 and 305 weapon skill are both big bumps in terms of mob difficulty.

That theory falls flat on its face as soon as you look at what happens when you let the gap rise even more, say you're 59 against 63. It makes **zero** sense for the weapon skill formula to make such an odd exception for one particular skill point, especially **when the following skill points are then much less penalized again**. If **every** point was 1.4% after that, nobody would have a problem with the formula, what I take a massive issue with is that 1.4% is fine for one point and one point only, but then every point past the threshold is 0.4% additional penalty instead.

**Comment by fgsfdss** [comment]
The .04 interval has been around since vanilla. Why change that of all things?

**Comment by latyy** [comment]
Have you considered how the rogue in this picture http://web.archive.org/web/20081111013822/http://sp00n.pytalhost.com/misc/backstabmiss.jpg
 while missing with 8% hit also is wearing what very much looks like Aged Core leather gloves (+5 dagger skill) http://db.vanillagaming.org/?item=18823 ? 
I mean it might be some other gloves, but surely there are not that many high level gloves with the same model as those.
According to the hitcap with 305 wepskill you posted earlier Cephel that should be impossible?

**Comment by Cephel** [comment]
>The .04 interval has been around since vanilla. Why change that of all things?

It wasn't changed. The change is for the lower part of the formula.

>According to the hitcap with 305 wepskill you posted earlier Cephel that should be impossible?

That would be impossible with **every** formula that I know of, including the one originally present on Elysium, the one I applied a patch for, the one I'm now suggesting and the one suggested by someone else further up the thread (the TBC one). +5 Skill should give you 2% hit in all of those formulas, so the simple answer is those cannot be the Aged Core Leather Gloves. If they are, it would change everything we know about weapon skill and would invalidate every single formula listed so far.

Another theory I have is, they are the Aged Core Gloves, however the Night Elf Rogue isn't the player, merely some other Rogue standing almost ontop of him, hiding the player.

**Comment by Rapter29** [comment]
>  Both formulas are speculative.

But one is more so than the other, at least the way I see it.

Also, @Cephel ,referring to your graphs, one thing is noticeable, in both the current and old formula, the slope before and after the gap remains the same. So other than the gap issue, it seems the rate of change per point should remain the same. However, your new formula changes the change per point after the gap.  

I think the main issue comes down to the fact that you object to the 1.4% jump. And while it is somewhat large, that in no way suggests that it is incorrect. 

The only real issue that DESERVES to be dealt with here is whether to start at 9% or 8.6%. So we should keep 'formula B (old)' as you put it, since it properly starts at 6% and ends at 5%. Then we decide on which formula A is more accurate. Voila. No need to re-invent the wheel.

**Comment by VeryRare** [comment]
I was referring to this table that was posted earlier:

![](https://i.gyazo.com/6b4fd2b80086747b0632519e747a590f.png)

Light's Hope is the current implementation that was committed by you. It's a .4 decrease that stops after 305. On any other formula that has ever been used in private servers or posted on those EJ threads the .4 decrease stops at 304.

If you are concerned about the gap why not make a new formula that does not need an if statement and could be scaled properly for each point? The original point of that If skillDiff line was created to MIMIC the gap from the available theory crafting evidence we have.

**Comment by knasen88** [comment]
Why would the difference between 4 and 5 wep skill be so huge on hit calculation but not on glancing, the old formula was definitely flawed because it made arbitrary jumps. Although the 5.6 miss at same skill level I agree would be weird aswell and the "new" proposed mix does look more reasonable.

**Comment by VeryRare** [comment]
I don't think you guys understand, this isn't a 'fun' server where we can make up our own game mechanics because they make the make the most sense to us. This is a blizzlike server intended to mimic the original retail release. 

The 'skillgap' is well documented and the product of the best testing data we have. Some of which can be seen here:  https://web.archive.org/web/20080108010959/http://elitistjerks.com:80/471121-post512.html

They even made their own calculator based on the formula: http://web.archive.org/web/20090727074431/http://sp00n.pytalhost.com:80/wow/weaponskill.htm

Further in the thread there is another post which claims to confirm these formulas:
http://web.archive.org/web/20090711164229/http://elitistjerks.com/f31/t11885-rogue_warrior_weapon_skill_adjustment_discussion/p22/#post472228
> When Def Skill - Weapon Skill <= 10:
> 5% + (Def Skill - Weapon Skill) * .1%

> When Def Skill - Weapon Skill > 10:
> 7% + (Def Skill - Weapon Skill - 10) * .4%

This is during TBC but it is the best source we have because there was no extensive research like this done for weapon skill during vanilla.  There's no indication this formula changed from vanilla to TBC and since we don't have any valid research from vanilla, the TBC formula is the closest we'll get to blizzlike hit chances

**Comment by Cephel** [comment]
>There's no indication this formula changed from vanilla to TBC 
Except the 3 times where it did. I have a blue post contradicting the formula linked in the EJ thread linked above. TBC research **is not valid**. There is a reason why TBC also saw the abolishment of the entire system. The system did not work in TBC, the values did not scale up the way they should and more likely than not, the gap is some weird artifact introduced either in TBC or as a consequence by moving from 60 to 70.

While there is no paper tests done on weapon skill, you can pick together some useful information by waybackmachine-ing the wowwiki during vanilla times and comparing retail logs like these:
http://oblistats.free.fr/stats/wws-history/

with BIS-lists from the old worldofwarcraft.com forums and BIS-lists from EJ (no longer online unfortunately), forums and some select guild websites.

Unless you can conclusively show me that there is a gap in the weapon skill formula **in Vanilla**, we should not introduce such a moronic mechanic.

**Comment by knasen88** [comment]
Explain why hit would be an arbitrary retarded Rollercoaster while glancing damage is like the current hit formula

**Comment by ohganot** [comment]
Glancing formula benefits mainly from the first 5 points too. I think we should use the TBC formula considering that vanilla had a hit cap bigger than 8%, so its likely that they didnt change the formula from 8.6% cap to 9% cap for no reason.

**Comment by ohganot** [comment]
I had an interesting idea. What if there wasn't a bump, but the scaling for the first 5 weapon skill was 0.6% instead of 0.4%? That would keep the 9% miss chance that was documented in TBC, but also get rid of the jump between 304 to 305 weapon skill. There were people who thought the formula was like that in TBC too.

| Weapon Skill | Miss Chance |
| --- | --- |
| 300 | 9% |
| 301 | 8.4% |
| 302 | 7.8% |
| 303 | 7.2% |
| 304 | 6.6% |
| 305 | 6% |
| 306 | 5.9% |
| 307 | 5.8% |
| 308 | 5.7% |
| 309 | 5.6% |
| 310 | 5.5% |
| 311 | 5.4% |
| 312 | 5.3% |
| 313 | 5.2% |
| 314 | 5.1% |
| 315 | 5% |

There are some sources that also support this:
http://wowwiki.wikia.com/wiki/Talk:Miss
"You're right I believe I probably should have double checked the math. It would be 0.1% for the first 10 and 0.6% for each point after that. However, it is untested beyond the first 15 points. I would need to find my notes before I make any other changes. Xer01 11:35, 11 September 2007"

https://web.archive.org/web/20071227195324/http://elitistjerks.com/f31/t11885-rogue_warrior_weapon_skill_adjustment_discussion/p20/
"Applying the above formula yields the following base miss rates for a Level 70 character with 350 Weapon Skill:
v. Level 70:  5.00%
v. Level 71:  5.50%
v. Level 72:  6.00%
v. Level 73:  9.00%

Because the formula is dependent upon the differential between Weapon Skill and Defense Skill, adding Weapon Skill (whether through talents, racial bonuses or gear) will reduce your miss rates against all mobs. But, because of the way the formula works, the beneficial impact of adding Weapon Skill is non-linear. In particular, as a Level 70 character, adding the first 5 levels of Weapon Skill (i.e., getting your Weapon Skill to 355) is the most beneficial, because it will effectively reduce your base miss rate against Level 73 mobs/bosses from 9% to 6%."

**Comment by knasen88** [comment]
Yes, the first 5 is bigger for glancing damage, but going from +4 to +5 wep skill does not give you more glancing damage than the first 4 points combined. You cannot seriously imply that would be the intended mechanic but rather blizz fucking around during TBC removal of +skill.

also that professional rogue with his gloves seems retarded, not only is 31/8/10 real fucking bad, it doesn't even utilize all talent points, his evidence is also "well duh prove to me that I did not see a miss during my shitter years" while providing jack shit for actual evidence

**Comment by ohganot** [comment]
"I used to save my Recaps for a lot of bosses, so I was able to confirm a lot of them".

If he noticed backstab misses without ACLG, he would obviously also notice them with ACLG.

**Comment by Cephel** [comment]
>. Attacking a lvl 62 mob has 6% spell miss chance, while attacking a lvl 63 mob has 17% spell miss chance.
Yes, and attacking a level 64 mob (or 63 if you're 59) has an even greater penalty, the penalty does not randomly go LESS severe after that.

The 0.1% -> 0.4% scaling _is_ the more severe case already, that's what the spell caster mechanic is mirrored as. The bump is an anomaly.

**Comment by ohganot** [comment]
Thats just your opinion. It's not an anomaly if it was documented in early TBC.

**Comment by Cephel** [comment]
It's not my opinion, your theory is just wrong. The bump is not a means to signify harder enemies, because that's not how it acts. If it was 1.4% for *every* point after that, then you'd have a leg to stand on.

**Comment by ohganot** [comment]
It's not a theory. It's already proven. Your opinion is a theory, which isn't proven yet. You need to find Vanilla sources that indicate that there isn't a bump, because currently we only have sources that indicate there being a bump.

**Comment by VeryRare** [comment]
First of all, please stop making up new formulas because they make sense to you.  .40% is used all through-out different vanilla game mechanics and that should never change for the first 4 skill points. It's not an "arbitrary retarded rollercoaster" - it's pretty simple.  .40% difference until raids start. At which point an increased in difficulty is needed. After that a change of .10% is used to reach the 5% miss rate.

As mentioned above Vanilla is full of idiosyncrasies and because a game mechanic does not make sense to you is not a valid argument for changing it. 

Notice that in all of Cephel's post he has never once linked any actual evidence. It's very interesting to me that the first change that was accepted by the Elysium team and was done so without any evidence. Almost as if he knew someone on the team and convinced them personally to make the change for his own benefit.

Secondly, the game mechanic makes perfect sense to me. 305 skill marks the start of raid encounters. An increased difficulty is needed and reflected in many other mechanics. 

So, because this game mechanic makes sense to me - According to Cephel I have just as much as a valid argument for why his change should be reverted back as he does for keeping it. Except it isn't.  Actual evidence is needed.

We have extremely valid research from TBC and years of private server experience that reflect a standard. This standard was changed without ANY evidence or logical reason and minimal input from the community by a different team and server. 

If this standard game mechanic that's been accepted by numerous different private server teams and the entire TBC theory-crafting community needs to be changed there must be some kind of evidence to warrant a deviance of this nature. 

At this point IMO the only discussion should be whether or not a 9% miss chance or 8.6% chance is the base. I prefer the TBC formula as it has the most evidence I've seen but since 8.6% miss effectively requires 9% hit to cap I am ok with either. 

Now, if for some wild reason the developers choose to use a new custom formula the entire function should be re-written.  It's a piece-wise function and by trying to remove the gap you are writing sloppy spaghetti code.  Write a new formula that does not use piece-wise and scales properly if so. Don't introduce more spaghetti code into the code base for your personal reasons or benefit.
